search/routes/api/serialize.py

```diff
@@ -122,31 +127,84 @@ def as_json(document_or_set: Union[DocumentSet, Document],
-#         'modified_date': '{%s}updated' % ATOM,
-#         'abstract': '{%s}summary' % ATOM,
-#         ''
-#     }
-#
-#     def __init__(cls, *args, **kwargs) -> None:
-#         super(AtomXMLSerializer, cls).__init__(*args, **kwargs)
-#         cls._root = etree.Element('feed', nsmap=cls.NSMAP)
-#
-#     def transform(cls):
-#         for document in cls.iter_documents():
-#
-#
-#
-#     def __repr__(cls) -> str:
-#         return etree.tostring(cls._root, pretty_print=True)
+    @classmethod
+    def transform_document(cls, fg: FeedGenerator, doc: Document,
+                           query: Optional[APIQuery] = None) -> None:
+        """Select a subset of :class:`Document` properties for public API."""
+        entry = fg.add_entry()
+        entry.id(url_for("api.paper", paper_id=doc['paper_id'],
+                             version=doc['version'], _external=True))
+        entry.title(doc['title'])
+        entry.summary(doc['abstract'])
+        entry.published(doc['submitted_date'])
+        entry.updated(doc['updated_date'])
+        entry.link({'href': url_for("api.paper", paper_id=doc['paper_id'], version=doc['version'], _external=True),
+                    "type": "text/html"})
+        #entry.link({'href': url_for("api.pdf", paper_id=doc['paper_id, version=doc['version, _external=True),
+        #            "type": "application/pdf", 'rel': 'related'})
+
+        if doc['comments']:
+            entry.arxiv.comment(doc['comments'])
+
+        if doc['journal_ref']:
+            entry.arxiv.journal_ref(doc['journal_ref'])
+
+        if doc['doi']:
+            entry.arxiv.doi(doc['doi'])
+        """
+        entry.arxiv.primary_category(doc['primary_classification'].archive['id'])
+        entry.category(
+            term=doc['primary_classification'].archive['id'],
+            scheme='http://arxiv.org://arxiv.org/schemas/atom')
+        
+        for category in doc['secondary_classification']:
+            entry.category(
+                term=category.archive['id'],
+                scheme='http://arxiv.org://arxiv.org/schemas/atom')
+        """
+        for author in doc['authors']:
+            author_data = {
+                "name": author['full_name']
+            }
+            if author.get('affiliation'):
+                author_data['affiliation'] = author['affiliation']
+            entry.arxiv.author(author_data)
+
+    @classmethod
+    def serialize(cls, document_set: DocumentSet,
+                  query: Optional[APIQuery] = None) -> str:
+        """Generate Atom response for a :class:`DocumentSet`."""
+        fg = FeedGenerator()
+        fg.register_extension('opensearch', OpenSearchExtension)
+        fg.register_extension("arxiv", ArxivExtension, ArxivEntryExtension, rss=False)
+        fg.id("http://api.arxiv.org/") # TODO: review API ID generation
+        fg.title(f"arXiv Query: {query}")
+        fg.link({"href" : "https://api.arxiv.org/", "type": 'application/atom+xml'})
```

Comment:
let's make this URL configurable

Reply:
This URL should be the URL of the request itself, and the code is in the `routes.api.serialize` module, so we should have access to the request.

Can you give me guidance on whether to implement that by just using the request object or whether to generate a URL given the APIQuery object?

Also noticing that perhaps query should not be an `APIQuery`, but rather a `Query`, which again raises the issue of the `ClassicAPIQuery` not inheriting properly.

Reply:
Gotcha. 

I would suggest extending the ``serialize()`` method to include a kwarg for the URL. So the route function would pluck the URL off the request and pass it in. This way you don't have to change ``APIQuery`` or the parameterization of the controller where the  is constructed.

Reply:
Looks like this still needs to be done

Reply:
Corrected! Using `url_for` to build the URL :)